src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -452,6 +482,54 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
     });
   }
 
+  handleValidationAndUpdateOfDeadlines(submissionEndTimestamp: number): Promise<{
+    isNotifyDeadlines: boolean, isAcceptDeletionOfDeadlines: boolean,
+   }> {
+    const affectedStudentDeadlines = DeadlineExtensionHelper.setDeadlinesBeforeEndTime(
+      this.studentDeadlines, submissionEndTimestamp);
+    const affectedInstructorDeadlines = DeadlineExtensionHelper.setDeadlinesBeforeEndTime(
+      this.instructorDeadlines, submissionEndTimestamp);
+    const affectedStudents = this.studentsOfCourse.filter((student) => affectedStudentDeadlines[student.email]);
+    const affectedInstructors = this.instructorsCanBePreviewedAs
+      .filter((instructor) => affectedInstructorDeadlines[instructor.email]);
```

Comment:
Perhaps we should try using `affectedStudentDeadlines.includes(student.email)`?

Reply:
I get what you mean, to make it clearer, but `affectedStudentDeadlines` is a record.  I'll change it to `affectedStudentDeadlines.hasOwnProperty(student.email)`

Reply:
Actually, because of https://eslint.org/docs/rules/no-prototype-builtins, the full fix will be:
`    const affectedStudents = this.studentsOfCourse.filter(
      (student) => Object.prototype.hasOwnProperty.call(affectedStudentDeadlines, student.email));
`
I think the original might actually be more readable. What do you think?

Reply:
Ah I see. I agree that the original is probably clearer. Let's keep that